charms/istio-pilot/lib/charms/istio_pilot/v0/ingress.py

```diff
@@ -0,0 +1,305 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""# Ingress Interface Library.
+"""
+
+import logging
+from functools import cached_property
+from pathlib import Path
+
+import yaml
+
+from ops.charm import CharmBase
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+from ops.model import BlockedStatus, WaitingStatus
+from serialized_data_interface import (
+    get_schema,
+    SerializedDataInterface,
+    NoCompatibleVersions,
+    NoVersionsListed,
+)
+
+logger = logging.getLogger(__name__)
+
+# The unique Charmhub library identifier, never change it
+LIBID = "b521889515b34432b952f75c21e96dfc"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+SCHEMA_URL_BASE = "https://raw.githubusercontent.com/canonical/operator-schemas"
+# SCHEMA_URL = f"{SCHEMA_URL_BASE}/master/ingress.yaml"
+SCHEMA_URL = f"{SCHEMA_URL_BASE}/1ed74c640bc289b71f261cda67177ee5209a1562/ingress.yaml"
```

Comment:
TODO: Remove once https://github.com/canonical/operator-schemas/pull/4 is merged and before this is.